src/utils.sh

```diff
@@ -454,16 +454,39 @@ function handlers_init() {
 
 export -f handlers_init
 
+function generate_uuid() {
+    source /etc/os-release
+    case $ID in
+        ubuntu)
+            if [ "$VERSION_ID" == "18.04" ];
```

Comment:
I don't think this check is needed. By the time we get here we already know that is is '18.04' :-)